trieste/space.py

```diff
@@ -171,3 +172,19 @@ def discretize(self, num_samples: int) -> DiscreteSearchSpace:
             this :class:`Box`.
         """
         return DiscreteSearchSpace(points=self.sample(num_samples))
+
+    def __mul__(self, box: Box):
+        """
+        Return the Cartesian product of the two :class:`Box`\ es (concatenating their respective lower and upper bounds).
+        :param box: :class:`Box`.
+        :return: the new combined :class:`Box`, or NotImplemented if respective bounds have different dtypes.
+        """
+        if self.lower.dtype is not box.lower.dtype:
+            return NotImplemented(
+                f"Bounds of both boxes must have the same dtype, got {self.lower.dtype} and"
+                f" {box.lower.dtype}"
+            )
```

Comment:
haha, that's a funny bug. `NotImplemented` doesn't take arguments (it's a builtin constant not an exception type). So when you try to pass it arguments, python raises a `TypeError`, and your test passes